cmd/client-fs.go

```diff
@@ -223,29 +223,42 @@ func (f *fsClient) Watch(ctx context.Context, options WatchOptions) (*WatchObjec
 }
 
 func preserveAttributes(fd *os.File, attr map[string]string) *probe.Error {
-	mode, e := strconv.ParseUint(attr["mode"], 10, 32)
-	if e != nil {
-		return probe.NewError(e)
-	}
+	if val, ok := attr["mode"]; ok {
+		mode, e := strconv.ParseUint(val, 10, 32)
+		if e != nil {
+			return probe.NewError(e)
+		}
 
-	// Attempt to change the file mode.
-	if e := fd.Chmod(os.FileMode(mode)); e != nil {
-		return probe.NewError(e)
+		// Attempt to change the file mode.
+		if e := fd.Chmod(os.FileMode(mode)); e != nil {
+			return probe.NewError(e)
+		}
 	}
 
-	uid, e := strconv.Atoi(attr["uid"])
-	if e != nil {
-		return probe.NewError(e)
+	var uid, gid int
+	var gidExists, uidExists bool
+	var e error
+	if val, ok := attr["uid"]; ok {
+		uid, e = strconv.Atoi(val)
+		if e != nil {
+			return probe.NewError(e)
+		}
+		uidExists = true
 	}
 
-	gid, e := strconv.Atoi(attr["gid"])
-	if e != nil {
-		return probe.NewError(e)
+	if val, ok := attr["gid"]; ok {
+		gid, e = strconv.Atoi(val)
+		if e != nil {
+			return probe.NewError(e)
+		}
+		gidExists = true
 	}
 
 	// Attempt to change the owner.
-	if e := fd.Chown(uid, gid); e != nil {
-		return probe.NewError(e)
+	if gidExists && uidExists {
+		if e := fd.Chown(uid, gid); e != nil {
```

Comment:
for `Chown`, we can pass `-1` for whichever value was *not* specified.
e.g if uid was *not* specified,
```go
if e := fd.Chown(-1, gid); e != nil {
...
```

Reply:
Ok.
Incorporated your comment.